apps/api/src/modules/lineups/lineups.controller.ts

```diff
@@ -0,0 +1,107 @@
+import { Request, Response, NextFunction } from "express";
+import { pool } from "../../database";
+import * as lineupsService from "./lineups.service";
+import { createLineupSchema, fixtureIdParamSchema } from "./lineups.validators";
+
+type Role = "COACH" | "PLAYER";
+type RequestUser = { id: number; email: string; role: Role };
+
+export async function saveLineup(
+  req: Request,
+  res: Response,
+  next: NextFunction
+) {
+  try {
+    const user = (req as any).user as RequestUser | undefined;
+    if (!user) {
+      return next(Object.assign(new Error("Unauthorized"), { status: 401 }));
+    }
+    if (user.role !== "COACH") {
+      return next(
+        Object.assign(new Error("Forbidden: Coach only"), { status: 403 })
+      );
+    }
+
+    const { id: fixtureId } = fixtureIdParamSchema.parse(req.params);
+    const body = createLineupSchema.parse(req.body);
+
+    const fixtureResponse = await pool.query(
```

Comment:
Your controller should not be handling the responsibility for querying the database. This should only be done in services.